backend/apps/slack/management/commands/slack_sync_data.py

```
@@ -41,90 +35,122 @@ def handle(self, *args, **options):
-                    if delay:
-                        time.sleep(delay)
-            except SlackApiError as e:
-                self.stdout.write(
-                    self.style.ERROR(f"Failed to fetch conversations: {e.response['error']}")
-                )
-            if conversations:
-                Conversation.bulk_save(conversations)
-                self.stdout.write(self.style.SUCCESS(f"Populated {total_channels} channels"))
+    def _call_slack_api(self, func, *args, **kwargs):
+        def _raise_max_retries():
+            raise RuntimeError("Max retries exceeded while calling Slack API")
```

Ruff (0.11.9)

Avoid specifying long messages outside the exception class

(TRY003)